File: 0271-encode-and-decode-strings/0271-encode-and-decode-strings.js
/**
 * Encodes a list of strings to a single string.
 *
 * @param {string[]} strs
 * @return {string}
 */
var encode = function(strs) {
    let answer = "";
    
    for (const str of strs) {
        answer += str.length;
        answer += "#";
        answer += str;
    }
    
    return answer;
};

/**
 * Decodes a single string to a list of strings.
 *
 * @param {string} s
 * @return {string[]}
 */
var decode = function(s) {
    const answer = [];
    
    let i = 0;
    while (i < s.length) {
        let j = i;
        while (s[j] !== "#") {
            j++;
        }
        const length = +s.substring(i, j);
        answer.push(s.substring(j + 1, j + 1 + length))
        i = j + 1 + length;
    }
    
    return answer;
};

/**
 * Your functions will be called as such:
 * decode(encode(strs));
 */

File: 0271-encode-and-decode-strings/0271-encode-and-decode-strings.java
public class Codec {
    // Encodes a list of strings to a single string.
    public String encode(List<String> strs) {
        StringBuilder sb = new StringBuilder();
        
        for (String str: strs) {
            sb.append(str.length());
            sb.append("#");
            sb.append(str);
        }
        
        return sb.toString();
    }

    // Decodes a single string to a list of strings.
    public List<String> decode(String s) {
        List<String> answer = new ArrayList<>();
        
        int i = 0;
        while (i < s.length()) {
            int j = i;
            while (s.charAt(j) != '#') {
                j++;
            }
            int length = Integer.parseInt(s.substring(i, j));
            answer.add(s.substring(j + 1, j + 1 + length));
            i = j + 1 + length;
        }
        
        return answer;
    }
}

// Your Codec object will be instantiated and called as such:
// Codec codec = new Codec();
// codec.decode(codec.encode(strs));